lib/xnvme_be_posix_async_thrpool.c

```diff
@@ -203,10 +203,12 @@ _posix_async_thrpool_term(struct xnvme_queue *q)
 		XNVME_DEBUG("FAILED: pthread_mutex_unlock(), err_lock: %d", err_lock);
 	}
 
-	for (int i = 0; i < queue->nthreads; i++) {
-		pthread_join(queue->threads[i], NULL);
+	if (queue->threads) {
+		for (int i = 0; i < queue->nthreads; i++) {
+			pthread_join(queue->threads[i], NULL);
+		}
+		free(queue->threads);
```

Comment:
Guarding the ``free()`` should not be needed.

Reply:
Done